internal/venice-common/src/main/resources/avro/KafkaMessageEnvelope/v13/KafkaMessageEnvelope.avsc

```diff
@@ -0,0 +1,383 @@
+                      "type": "long"
+                    }
+                  ]
+                }, {
+                  "name": "VersionSwap",
+                  "doc": "This controlMessage is written to the real-time topic by the controller or to the store-version topic by the current version's leader server. It can be used to let current version and future version synchronize on a specific point for all regions' real-time topics, to guarantee there is only one store version producing to change capture topic all the time. It can also be used by the consumer client to switch to another store-version topic and filter messages that have a lower watermark than the one dictated by the leader.",
+                  "type": "record",
+                  "fields": [
+                    {
+                      "name": "oldServingVersionTopic",
+                      "doc": "Name of the old source topic we are switching from.",
+                      "type": "string"
+                    }, {
+                      "name": "newServingVersionTopic",
+                      "doc": "Name of the new source topic we are switching to.",
+                      "type": "string"
+                    }, {
+                      "name": "localHighWatermarks",
+                      "doc": "The latest offsets of all real-time topic has been consumed up until now.",
+                      "type": [
+                        "null",
+                        {
+                          "type": "array",
+                          "items": "long"
+                        }
+                      ],
+                      "default": null
+                    }, {
+                      "name": "localHighWatermarkPubSubPositions",
+                      "doc": "The latest pubsub positions of all real-time topics consumed up until now.",
+                      "type": {
+                        "type": "array",
+                        "items": "bytes"
+                      },
+                      "default": []
+                    }, {
+                      "name": "isRepush",
+                      "doc": "Flag to indicate this version swap is triggered by repush or not.",
+                      "type": "boolean",
+                      "default": false
+                    }, {
+                      "name": "isLastVersionSwapMessageFromRealTimeTopic",
+                      "doc": "Flag to indicate this version swap message in version topic is triggered by the last version swap in real time topic the leader server has received. With this flag, new leader will be able to recover the full state during leadership handover, when we rely on real-time topics for all regions to achieve version swap synchronization.",
+                      "type": "boolean",
+                      "default": false
+                    }, {
+                      "name": "sourceRegion",
```

Comment:
> To ensure no data gaps in true A/A we need to consume all version swaps with different destinationRegion

I have a dumb question - will log compaction compact the VS message so that you only see the latest one? And thus a newly started consumer may not swap?